File: app/src/main/java/com/xeniac/warrantyrostermanager/database/CategoryDAO.java
package com.xeniac.warrantyrostermanager.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.xeniac.warrantyrostermanager.model.Category;

import java.util.List;

@Dao
public interface CategoryDAO {

    @Insert
    void insertAllCategories(List<Category> categoriesList);

    @Insert
    void insertCategory(Category... category);

    @Query("DELETE FROM Category")
    void deleteAllCategories();

    @Delete
    void deleteCategory(Category... category);

    @Query("SELECT COUNT(*) FROM Category")
    int countItems();

    @Query("SELECT * FROM Category ORDER BY title")
    List<Category> getAllCategories();

    @Query("SELECT title FROM Category ORDER BY title")
    List<Integer> getAllCategoryTitles();

    @Query("SELECT * FROM Category WHERE id = :categoryId")
    Category getCategoryById(String categoryId);
}

File: app/src/main/java/com/xeniac/warrantyroster_manager/db/CategoryDao.kt
package com.xeniac.warrantyroster_manager.db

import androidx.room.*
import com.xeniac.warrantyroster_manager.model.Category

@Dao
interface CategoryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAllCategories(categoriesList: List<Category>)

    @Query("DELETE FROM categories")
    suspend fun deleteAllCategories()

    @Query("SELECT COUNT(*) FROM categories")
    fun countItems(): Int

    @Query("SELECT * FROM categories")
    fun getAllCategories(): List<Category>

    @Query("SELECT * FROM categories WHERE id = :categoryId")
    fun getCategoryById(categoryId: String): Category
}